server/src/main/java/org/apache/druid/metadata/SqlSegmentsMetadataQuery.java

```diff
@@ -286,7 +286,7 @@ private List<DataSegmentPlus> retrieveSegmentBatchById(
     if (includeSchemaInfo) {
       final Query<Map<String, Object>> query = handle.createQuery(
           StringUtils.format(
-              "SELECT payload, used, schema_fingerprint, num_rows FROM %s WHERE dataSource = :dataSource %s",
+              "SELECT payload, used, upgraded_from_segment_id, schema_fingerprint, num_rows FROM %s WHERE dataSource = :dataSource %s",
```

Comment:
Nit: Put the new column last in the list of selected columns

Reply:
Reverted order